apps/system/js/pin_page_system_dialog.js

```diff
@@ -169,25 +169,16 @@
 
     Service.request('Places:isPinned', data.url)
       .then((isPinned) => {
-        if (isPinned) {
-          this.pinPageButton.setAttribute('data-l10n-id', 'pinning-unpin-page');
-          this.pinPageButton.dataset.action = 'unpin-page';
-        } else {
-          this.pinPageButton.setAttribute('data-l10n-id', 'pinning-pin-page');
-          this.pinPageButton.dataset.action = 'pin-page';
-        }
+        var action = isPinned ? 'unpin-page' : 'pin-page';
+        this.pinPageButton.setAttribute('data-l10n-id', 'pinning-' + action);
+        this.pinPageButton.dataset.action = action;
       });
 
-    Service.request('PinsManager:isPinned', data.url)
-      .then((isPinned) => {
-        if (isPinned) {
-          this.pinSiteButton.setAttribute('data-l10n-id', 'pinning-unpin-site');
-          this.pinSiteButton.dataset.action = 'unpin-site';
-        } else {
-          this.pinSiteButton.setAttribute('data-l10n-id', 'pinning-pin-site');
-          this.pinSiteButton.dataset.action = 'pin-site';
-        }
-      });
+    BookmarksDatabase.get(data.url).then((site) => {
```

Comment:
We should still set the button to "unpin site" for all URLs within the scope of a pinned site which specifies a real "scope" in its manifest.